crypto/ocsp/ocsp_integration_test.cc

```diff
@@ -181,31 +205,48 @@ TEST_P(OCSPIntegrationTest, AmazonTrustServices) {
   ASSERT_TRUE(SSL_library_init());
   bssl::UniquePtr<SSL_CTX> ssl_ctx(SSL_CTX_new(TLS_method()));
   ASSERT_TRUE(ssl_ctx);
+
+  // Set up trust store.
+  bssl::UniquePtr<X509_STORE> trust_store(SetupTrustStore());
+  ASSERT_TRUE(SSL_CTX_set1_verify_cert_store(ssl_ctx.get(), trust_store.get()));
   bssl::UniquePtr<BIO> bio(BIO_new_socket(sock, BIO_CLOSE));
   bssl::UniquePtr<SSL> ssl(SSL_new(ssl_ctx.get()));
+  // Set the host name, so we don't retrieve the wrong intended certificate
+  // chain.
```

Comment:
NP: this has more to do with letting the endpoint know what we're trying to connect to, and in this case SNI is used to host multiple different websites on one server.

Reply:
Re-clarified the comment